### ISSUE TITLE
Advanced Search - Columns

### ISSUE THREAD

**Issue by mlipok** [osTicket/osTicket#5259]
### Prerequisites

* [ Y ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ N ] Do you have any errors in the PHP error log, or javascript console?
* [ Y ] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [ Y ] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description
"Advanced Search form" have a tab "Columns"
You could add/delete tab, but the result of searching will not return results related to the selected "Columns"

[Description of the bug or feature]

### Steps to Reproduce

1. Go to "Open"
2. Click "Advanced Search"
3. Add some criteria
4. Click the tab "Columns"

**Expected behavior:** [What you expected to happen]
If you Choose "OPEN" or "CLOSED" or any other "Parent/TopLevel Queue" from "---My searches---" ComboBox, the function "Use standard columns" should be checked the column list should be possible to change in the same way, how it works when you add SubQueries in:
Admin Panel >> Settings >> Tickets >> Ticket Queues  >> Add New Queue

I mean the columns should be inherited from "Parent/TopLevel Queue" and you should be able to add new/additional columns.

Attached Screenshot is showing what is wrong.
![AdvanceSearch_Column](https://user-images.githubusercontent.com/11089482/70548506-f3e03400-1b72-11ea-8849-0e46168a3057.png)

**Actual behavior:** [What actually happened]
The result always shows a list with column:    Number, Created, Subject, From, Priority, Assignee

Never shows the expected/selected columns.
Even if you Choose "OPEN" or "CLOSED" or any other subqueries, from "---My searches---" ComboBox, the feature "Use standard columns" is always Checked.

Regardless of the above, you can add an additional column but it will not give you any effect on the result list - the columns on the list are always the same.

### Versions

Admin panel -> Dashboard -> Information which also additionally gives you information about your server.
v1.14.1 (f1e9e88)

--

**Comment by JediKev** [comment]
@mlipok 

Look at the top of the "Columns" tab in the Advanced Search popup and you will see that the "Use standard columns" setting is Disabled (meaning you can't change the value). Yea, we are supposed to hide the custom columns altogether for Advanced Searches. You are only supposed to customize the columns for Saved Searches, Personal Queues, Child Queues, and/or Parent Queues. I'm currently creating a fix for this but have ran into some other Search/Queue issues so I'm going to resolve those first.

Cheers.

**Comment by mlipok** [comment]
I would like to see a feature to preset columns for advance search or at least please use the same collumns and sort order like in OPEN TICKET I mean the same column and order like in ..../scp/tickets.php?queue=1

as I have strongly modified column for the queue=1 and I use them as a default column set and order.

**Comment by JediKev** [comment]
@mlipok 

>I would like to see a feature to preset columns for advance search or at least please use the same collumns and sort order like in OPEN TICKET

We use the legacy columns (the standard columns before Custom Columns feature was introduced) for Advanced Searches. If you would like to change this you can Save the search so you can modify the columns or modify the codebase here to input your own standard columns:
- https://github.com/osTicket/osTicket/blob/develop/include/class.queue.php#L679-L731

Cheers.

**Comment by protich** [comment]
@JediKev - I think feedback by @mlipok is something we should seriously consider. In user preference, like default queue, we can add queue selection to use for advanced search columns template.

**Comment by JediKev** [comment]
Right, I should've put `but we will add this to our list` at the end as I didn't mean that we wouldn't consider this..

**Comment by mlipok** [comment]
Thanks.

If I can ..... I want to make one remark.
As I wrote before:

> Even if you Choose "OPEN" or "CLOSED" or any other subqueries, from "---My searches---" ComboBox, the feature "Use standard columns" is always Checked


In this case this should be taken not from "default" queue, or even from user preference....
but this should be taken from the specified queue selected in from "---My searches---" ComboBox.

**Comment by host138** [comment]
https://forum.osticket.com/d/93777-change-columns-search-results
